server/backends/disk_cache/disk_cache.go

```diff
@@ -98,6 +98,8 @@ func NewDiskCache(rootDir string, maxSizeBytes int64) (*DiskCache, error) {
 }
 
 func (c *DiskCache) WithPrefix(prefix string) interfaces.Cache {
+	c.mu.Lock()
```

Comment:
I don't think this is sufficient.
The new DiskCache instance will capture the current value of c.diskIsMapped. 
If the value is false, it will never be updated to true on the new instance, only on the original cache.
That means the new cache can still attempt to write to fileChannel after it's closed.